lib/rom/adapter/lint/test.rb

```diff
@@ -0,0 +1,54 @@
+module ROM
+  class Adapter
+    module Lint
+      module TestAdapter
```

Comment:
@solnic the `TestAdapter` is precisely what we discussed. It's great. What a help to developers to be able to wire this up and see if their adapter meets the requirements.